From 60948466dec9727c8040e3de45cb15a71ac3967f Mon Sep 17 00:00:00 2001
From: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Date: Tue, 31 May 2016 16:59:13 -0300
Subject: [PATCH 61/88] staging/android: add DEBUG_FS dependence on Kconfig
Git-commit: b1f656044a9936880d5a7460cd8784b2e124ab87
Patch-mainline: v4.8-rc1
References: fate#321223

SW_SYNC only works with DEBUG_FS so state it in the Kconfig file.

Signed-off-by: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Reviewed-by: Sumit Semwal <sumit.semwal@linaro.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/staging/android/Kconfig      | 1 +
 drivers/staging/android/sync_debug.c | 4 ----
 drivers/staging/android/sync_debug.h | 4 +---
 3 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/staging/android/Kconfig b/drivers/staging/android/Kconfig
index 7d13a127dcdc..f4d86780ff50 100644
--- a/drivers/staging/android/Kconfig
+++ b/drivers/staging/android/Kconfig
@@ -42,6 +42,7 @@ config SW_SYNC
 	bool "Software synchronization framework"
 	default n
 	depends on SYNC_FILE
+	depends on DEBUG_FS
 	---help---
 	  A sync object driver that uses a 32bit counter to coordinate
 	  synchronization.  Useful when there is no hardware primitive backing
diff --git a/drivers/staging/android/sync_debug.c b/drivers/staging/android/sync_debug.c
index b76022623d8a..903296988d41 100644
--- a/drivers/staging/android/sync_debug.c
+++ b/drivers/staging/android/sync_debug.c
@@ -17,8 +17,6 @@
 #include <linux/debugfs.h>
 #include "sync_debug.h"
 
-#ifdef CONFIG_DEBUG_FS
-
 static struct dentry *dbgfs;
 
 static LIST_HEAD(sync_timeline_list_head);
@@ -225,5 +223,3 @@ void sync_dump(void)
 		}
 	}
 }
-
-#endif
diff --git a/drivers/staging/android/sync_debug.h b/drivers/staging/android/sync_debug.h
index 48e2d1c8c3ef..425ebc5c32aa 100644
--- a/drivers/staging/android/sync_debug.h
+++ b/drivers/staging/android/sync_debug.h
@@ -41,9 +41,7 @@ struct sync_timeline {
 
 	struct list_head	active_list_head;
 
-#ifdef CONFIG_DEBUG_FS
 	struct list_head	sync_timeline_list;
-#endif
 };
 
 static inline struct sync_timeline *fence_parent(struct fence *fence)
@@ -64,7 +62,7 @@ struct sync_pt {
 	struct list_head active_list;
 };
 
-#ifdef CONFIG_DEBUG_FS
+#ifdef CONFIG_SW_SYNC
 
 extern const struct file_operations sw_sync_debugfs_fops;
 
-- 
2.12.0
